### ISSUE TITLE
No events are being logged in Google Analytics

### ISSUE THREAD

**Issue by the4dpatrick** [lwansbrough/react-native-google-analytics#21]
This is a separate issue from #19. 

Events are not consistently being logged into Google Analytics.
I'm not completely sold that the issue is a result of using this package. 

When I try to enter this url
`https://ssl.google-analytics.com/collect?v=1&tid=<TRACKING_ID>&cid=5555&t=screenview`

into the browser while looking at the google analytics' realtime view, I am unable to see the events being tracked. Some times events are being tracked however. I've tried switching out the `TRACKING_ID` with websites' tracking id and not an apps, and an event seems to be tracked. The only thing is that the event tracking is not consistent.

Possibilities for debugging:
    \* Try different query params
    \* Try different Tracking IDs
    \* Try different User Agents/devices

**Comment by rmevans9** [comment]
Has there been any progress made to figuring out why this is happening? I setup this component today and got a handful of hits then it went to nothing while I was trying to test. I confirmed the fetch call is returning a success, just nothing showing in GA.

**Comment by lwansbrough** [comment]
Can you confirm you're using the latest version of the module? This looks like it might be an issue with caching, which was solved (to the best of my knowledge) in `0.1.0`.

**Comment by lwansbrough** [comment]
You should be seeing a `&z=` parameter in the requests coming from this module.

**Comment by rmevans9** [comment]
@lwansbrough I am on 0.1.0 and there is the &z parameter in the request. 

I tried troubleshooting a bit by way of switching from a GET request to a POST request as it appears GA accepts either which didn't have any positive results.

**Comment by malinthe** [comment]
So I tried to catch any errors on the fetch method used on Analytics.js and received the following error: `TypeError: Network request failed` Is this related to the issue?

**Comment by danleveille** [comment]
I had this problem and found that the issue was the `cd=` parameter was not being included.

The [cd parameter is required for screenview hits](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters), so the example listed in the README (`var screenView = new GAHits.ScreenView('GA Test', '1', 'com.example.app');` wouldn't work, right?

@the4dpatrick Your original request in #19 doesn't include a 5th parameter (`cd`) -- could that be your issue too? That would be why your website tracking ID would work (since `cd` isn't required for websites).

**Comment by the4dpatrick** [comment]
@danleveille 

I just tried another request `https://ssl.google-analytics.com/collect?v=1&tid=<GA_ID>&cid=555adfasd5&t=screenview&cd=HighScore` and still no luck.

So you've tested that events were being tracked by adding the `cd=` param to a request?

**Comment by danleveille** [comment]
Yeah, when I remove the `cd` parameter, it doesn't record the event in GA. When I add it back in, it tracks properly. 

Make sure you have all the required parameters. These parameters are all required: `v, an, tid, cid, cd, t`

**Comment by the4dpatrick** [comment]
It works! Good job!
`https://ssl.google-analytics.com/collect?v=1&tid=<GA_ID>&cid=5555&t=screenview&cd=foo&an=name`

Would you like to make the fixing PR? @danleveille

**Comment by danleveille** [comment]
@the4dpatrick Nice! Glad to hear. I'm honestly a bit of a newbie with Github (and React), so I'll leave the PR to someone else. :stuck_out_tongue_closed_eyes:

**Comment by the4dpatrick** [comment]
That's a perfect reason to make your first Pull Request! I can help and write down each step with the end result being your first PR. I'd be more than happy with getting you started in open source :)

I'm away from my computer now, but I'll can get the steps to you later today.
@danleveille

**Comment by lwansbrough** [comment]
Nice work guys, if someone could make that PR that would be greatly appreciated!

**Comment by danleveille** [comment]
@the4dpatrick Haha. Good point! I'll give it a shot. What should the actual fix should be?

In ScreenView.js:

``` javascript
constructor(appName, appVersion, appId, appInstallerId, screenName, experiment) {
```

I'm thinking the first two args should be `appName` (`an`) and `screenName` (`cd`), since those two args are the only ones required by GA. The others are optional.

Is that a good solution?

**Comment by lwansbrough** [comment]
Yup that looks like the correct solution to me.

**Comment by the4dpatrick** [comment]
Looks good.

I realize my mistake was that I was not including the 4th arg for `screenName`. I'd be sure to update the README too @danleveille

**Comment by edwardofclt** [comment]
Events are being sent again?

**Comment by the4dpatrick** [comment]
I haven’t used this repo for a while, but I can take a look at what I did. I’m don’t remember if by the end it worked or not.

**Comment by peacechen** [comment]
I've had this library running in our app for a year and GA log data isn't being captured.  Well sort of.  In Dev mode, *some* data is sent randomly, but very rarely.  No real-time data ever shows up.  There's no data at all for production mode.  I've set up the GA account with a Website property and added a Mobile view.

Here's a sample GA hit (split lines on the `&` to make it easier to read):
```
https://www.google-analytics.com/collect?
t=screenview&
an=Mobile&
cd=Router%20navigation&
av=1.2.3.1-131&
aid=com.company.appname&
v=1.2.3.1-131&
tid=UA-xxxxxxxx-7&
cid=<crazy_long_GUID_here>&
z=43954641
```

Reactotron logs those queries with a response of 200 so I'm confident that they're being sent.  When the network connection is flaky, I've received network error reports for those queries from our crash analysis tool. 

EDIT: Thanks to @the4dpatrick 's reference to the [Google's Measurement Protocol Reference](https://developers.google.com/analytics/devguides/collection/protocol/v1/reference), I noticed that the `v` param was incorrectly set to my app's version when it should always be `1`.  It's the 3rd parameter to the `Analytics` constructor:
```
ga = new Analytics(trackingID, clientId, 1, userAgent);
```
